module2/Homework2-start/ColorPicker/ContentView.swift

```diff
@@ -0,0 +1,167 @@
+import SwiftUI
+
+struct ContentView: View {
+  @Environment(\.verticalSizeClass) var verticalSizeClass
+  @Environment(\.horizontalSizeClass) var horizontalSizeClass
+
+  @State private var alertIsVisible: Bool = false
+  @State private var redColor: Double = Constants.defaultRedColor
+  @State private var greenColor: Double = Constants.defaultGreenColor
+  @State private var blueColor: Double = Constants.defaultBlueColor
+
+  // When the app first runs, the default for previewColor shows up as white and not these values. Why?
+  @State private var previewColor: Color =  Color(red: Constants.defaultRedColor, green: Constants.defaultGreenColor, blue: Constants.defaultBlueColor)
+  
+  var body: some View {
+    
+    ZStack {
+      // The size class vars are not getting set and evaluating this if-statement. Why?
+      if verticalSizeClass == .compact && horizontalSizeClass == .regular {
```

Comment:
Switch your simulator to one of the PRO MAX models and your code will work as written. A better if statement would be to just watch the `horizontalSizeClass` because of the wierd way that apple implements these. This is a pretty good blog post with diagrams. https://useyourloaf.com/blog/size-classes/

Reply:
Ok wow - so Pro version iPhones are actually normal size compared to Max and Plus. Now the simulator and previews make sense.

Reply:
Pro phones have nicer cameras. Max and plus have bigger screens. The whole size class thing is kind of a mess.